base/uk.ac.stfc.isis.ibex.ui.devicescreens.tests/src/uk/ac/stfc/isis/ibex/ui/devicescreens/tests/DeviceScreensDescriptionViewModelTest.java

```diff
@@ -134,55 +135,97 @@ public void constructor_produces_deep_copy_of_description() {
     @Test
     public void changing_the_selected_screen_updates_name_key_etc() {
         // Act
-        viewModel.setSelectedScreen(1);
+        List<DeviceDescriptionWrapper> first = viewModel.getScreens().subList(1, 2);
+        viewModel.setSelectedScreens(first);
         
         // Assert
-        assertEquals(deviceName + "2", viewModel.getCurrentName());
+        assertEquals(deviceName + "2", viewModel.getName());
         assertEquals(opiName1, viewModel.getCurrentKey());
-        assertEquals(opiDescription1, viewModel.getCurrentDescription());
-        assertEquals(deviceName + "2", viewModel.getSelectedScreen().getName());
+        assertEquals(opiDescription1, viewModel.getDescription());
     }
 
     @Test
     public void clearing_the_selected_screen_unsets_name_key_etc() {
         // Act
-        viewModel.setSelectedScreen(-1);
+        viewModel.setSelectedScreens(null);
 
         // Assert
-        assertEquals("", viewModel.getCurrentName());
+        assertEquals("", viewModel.getName());
         assertEquals("", viewModel.getCurrentKey());
```

Comment:
Why has "Current" been removed from other methods but not this one? On examination of EditDeviceScreensDescriptionViewModel, it doesn't track previous keys. Would using standard naming of getKey() and setKey() be more appropriate?